vector/src/main/java/im/vector/app/features/crypto/recover/BootstrapSharedViewModel.kt

```diff
@@ -127,28 +143,36 @@ class BootstrapSharedViewModel @AssistedInject constructor(
+                    val version = awaitCallback {
                         session.cryptoService().keysBackupService().getCurrentVersion(it)
                     }.toKeysVersionResult()
                     if (version == null) {
                         // we just resume plain bootstrap
                         doesKeyBackupExist = false
                         setState {
-                            copy(step = BootstrapStep.FirstForm(keyBackUpExist = doesKeyBackupExist, methods = this.secureBackupMethod))
+                            copy(step = BootstrapStep.FirstForm(keyBackUpExist = false, methods = this.secureBackupMethod))
                         }
                     } else {
-                        // we need to get existing backup passphrase/key and convert to SSSS
-                        val keyVersion = awaitCallback<KeysVersionResult?> {
-                            session.cryptoService().keysBackupService().getVersion(version.version, it)
-                        }
-                        if (keyVersion == null) {
-                            // strange case... just finish?
-                            _viewEvents.post(BootstrapViewEvents.Dismiss(false))
-                        } else {
-                            doesKeyBackupExist = true
-                            isBackupCreatedFromPassphrase = keyVersion.getAuthDataAsMegolmBackupAuthData()?.privateKeySalt != null
+                        if (backupMigrationStrategy == BackupMigrationStrategy.IgnoreExistingBackup) {
+                            clearAllExistingBackups()
+                            doesKeyBackupExist = false
                             setState {
-                                copy(step = BootstrapStep.FirstForm(keyBackUpExist = doesKeyBackupExist, methods = this.secureBackupMethod))
+                                copy(step = BootstrapStep.FirstForm(keyBackUpExist = false, methods = this.secureBackupMethod))
+                            }
+                        } else {
+                            // we need to get existing backup passphrase/key and convert to SSSS
+                            val keyVersion = awaitCallback<KeysVersionResult?> {
+                                session.cryptoService().keysBackupService().getVersion(version.version, it)
+                            }
+                            if (keyVersion == null) {
+                                // strange case... just finish?
+                                _viewEvents.post(BootstrapViewEvents.Dismiss(false))
+                            } else {
+                                doesKeyBackupExist = true
+                                isBackupCreatedFromPassphrase = keyVersion.getAuthDataAsMegolmBackupAuthData()?.privateKeySalt != null
+                                setState {
+                                    copy(step = BootstrapStep.FirstForm(keyBackUpExist = doesKeyBackupExist, methods = this.secureBackupMethod))
```

Comment:
It does not make big difference but since `doesKeyBackupExist` is mutable should we pass `true` to `keyBackUpExist` when updating the state instead?